src/redux/actions/internalActions.js

```diff
@@ -233,10 +233,16 @@ export const setOutlines = outlines => ({
   type: 'SET_OUTLINES',
   payload: { outlines },
 });
-export const setLayers = layers => ({
-  type: 'SET_LAYERS',
-  payload: { layers },
-});
+export const setLayers = layers => (dispatch) => {
+  dispatch({
+    type: 'SET_LAYERS',
+    payload: { layers },
+  });
+  const doc = core.getDocument();
+  doc.setLayersArray(layers);
+  window.docViewer.refreshAll();
+  window.docViewer.updateView();
```

Comment:
Do you think we should put this logic in a `componentDidUpdate` in `LayersPanel.js`? I guess that doesn't really make a difference...

Reply:
Done